semtypes/src/main/java/io/ballerina/types/Core.java

```diff
@@ -120,7 +120,7 @@
                 int c = code.code;
                 all = UniformTypeBitSet.from(all.bitset | 1 << c);
             } else {
-                subtypes.add(UniformSubtype.from(code, data));
+                subtypes.add(UniformSubtype.from(code, (ProperSubtypeData) data));
```

Comment:
Shall we retain these comments https://github.com/ballerina-platform/nballerina/commit/86ecd997d3031cc04e4e3d453cad6a5d2cabad78#diff-5008fa7817f6124661e3e735e6673f37538af2ab4de42527dc7c030ceea9afc2R519